api/contents.go

```diff
@@ -124,6 +132,60 @@ func ConfigureContentsRouter(e *echo.Group, dldm *core.DeltaDM) {
 
 		return c.JSON(http.StatusOK, results)
 	})
+
+	contents.POST("", func(c echo.Context) error {
+		var content []ContentCollection
+		results := struct {
+			Success []string `json:"success"`
+			Fail    []string `json:"fail"`
+		}{
+			Success: make([]string, 0),
+			Fail:    make([]string, 0),
+		}
+
+		if err := c.Bind(&content); err != nil {
+			return err
+		}
+
+		cache := make(map[string]uint)
+
+		for _, cnt := range content {
+			var dataset db.Dataset
+			// Check for bad data
+			if cnt.CommP == "" || cnt.PayloadCID == "" || cnt.PaddedSize == 0 || cnt.Size == 0 || cnt.Collection == "" {
+				results.Fail = append(results.Fail, cnt.CommP)
+				continue
+			}
+
+			// Check if dataset exists
+			if cache[cnt.Collection] == 0 {
+				if dldm.DB.Model(&db.Dataset{}).Where("name = ?", cnt.Collection).First(&dataset).Error != nil {
+					results.Fail = append(results.Fail, cnt.CommP)
```

Comment:
could you add some `log.Debug()` statements around these failures so we can examine debug logs in case they are failing?

Reply:
done... we could also change the return map to include these errors? Basically commp -> err relation under the fail array. Let me know.